src/core1/DynamicDoubleArray.java

```diff
@@ -30,22 +27,25 @@ public void add(double e) {
     }
 
     public void remove(int i) {
-        int t = 0;
-        double[] newArray = new double[currLength - 1];
-        for (int j = 0; j < currLength; j++) {
-            if (j != i){
-                newArray[t] = array[j];
-                t++;
-            }
-        }
+        System.arraycopy(array, i, array, i + 1, currLength - i - 1);
```

Comment:
I'm not pretty sure what happens if "i" would be negative or higher than array's lenght. So it's better to add some checks  to the methods and throw an Exception (for example, IndexOutOfBound) in wrong cases.